protocol.go

```diff
@@ -85,11 +85,16 @@ func (t tcpDialer) DialSqlConnection(ctx context.Context, c *Connector, p *msdsn
 	} else {
 		ips = []net.IP{ip}
 	}
-	if len(ips) == 1 {
-		d := c.getDialer(p)
-		addr := net.JoinHostPort(ips[0].String(), strconv.Itoa(int(resolveServerPort(p.Port))))
-		conn, err = d.DialContext(ctx, "tcp", addr)
-
+	if len(ips) == 1 || !p.MultiSubnetFailover {
+		// Try to connect to IPs sequentially until one is successful per MultiSubnetFailover false rules
+		for _, ipaddress := range ips {
+			d := c.getDialer(p)
+			addr := net.JoinHostPort(ipaddress.String(), strconv.Itoa(int(resolveServerPort(p.Port))))
+			conn, err = d.DialContext(ctx, "tcp", addr)
+			if err == nil {
```

Comment:
>if err == nil { [](http://example.com/codeflow?start=3&length=15)

Is there a difference in how errors from the multiple dial attempts are reported?
Does the error channel approach in the parallel implementation return an `error` object that aggregates all the parallel dial errors? If so, it may be desirable to perform a similar aggregation in the serial case.

Reply:
my quick hack implies that the parallel model would seemingly return a single random dial error from the list.

eg this sample just prints "2"

```go
package main

import (
 "fmt"
)

func main() {
 errChan := make(chan error, 2)
 go func() {
  errChan <- fmt.Errorf("1")
 }()
 go func() {
  errChan <- fmt.Errorf("2")
 }()
 var err error
 select {
 case err = <-errChan:
 }
 fmt.Println(err.Error())

}
```